processor/processor.go

```diff
@@ -1168,25 +1209,42 @@ func (proc *HandleT) processJobsForDest(subJobs subJob, parsedEventList [][]type
 				uniqueMessageIds[messageId] = struct{}{}
 				//We count this as one, not destination specific ones
 				totalEvents++
-				eventsByMessageID[messageId] = types.SingularEventWithReceivedAt{SingularEvent: singularEvent, ReceivedAt: receivedAt}
+				eventsByMessageID[messageId] = types.SingularEventWithReceivedAt{
+					SingularEvent: singularEvent,
+					ReceivedAt:    receivedAt,
+				}
 
-				//Getting all the destinations which are enabled for this
-				//event
-				backendEnabledDestTypes := getBackendEnabledDestinationTypes(writeKey)
-				enabledDestTypes := integrations.FilterClientIntegrations(singularEvent, backendEnabledDestTypes)
 				sourceForSingularEvent, sourceIdError := getSourceByWriteKey(writeKey)
 				if sourceIdError != nil {
 					proc.logger.Error("Dropping Job since Source not found for writeKey : ", writeKey)
 					continue
 				}
 
+				commonMetadataFromSingularEvent := makeCommonMetadataFromSingularEvent(
+					singularEvent,
+					batchEvent,
+					receivedAt,
+					sourceForSingularEvent,
+				)
+
+				//REPORTING - GATEWAY metrics - START
```

Comment:
The current implementation of capturing reporting metrics seems to be convoluted. The least I can say is that it is hard to follow, since as it appears many things need to happen at different stages of this method, without a clear purpose.
I will invest some time to study the current implementation in hope of forming a cleaner picture for the specific domain, otherwise I cannot be of any more use with regards to reviewing this PR.